addons/website_sale/controllers/main.py

```diff
@@ -271,7 +280,37 @@ def shop(self, page=0, category=None, search='', ppg=False, **post):
         if category:
             url = "/shop/category/%s" % slug(category)
 
-        product_count = len(search_product)
+        compute_currency = False
+        if search_product:
+            compute_currency = self._get_compute_currency_product_to_pricelist(pricelist, search_product[:1])
+
+            where, args = expression.expression(domain, Product).to_sql()
+            query = 'SELECT MIN("list_price") as min_list_price, MAX("list_price") as max_list_price FROM "product_template" WHERE %s'
```

Comment:
Not sure if this has been discussed with colleagues, I guess we don't have the choice without killing the perfs, but that won't reflect reality as pricelist won't be applied and could **completely** change the min/max prices.

Reply:
We should think of a way to include pricelist in prices, or it will only work with dummy data with no pricelists IMHO.

In real case, even a simple pricelist will break results. I just created a 50% discount for all products and this is the result, everything is wrong, max price, min price, max available price and min available price:

![pl_50_pc](https://user-images.githubusercontent.com/30048408/63338527-420ec480-c343-11e9-8716-c4bc1b2332d3.png)
-----
![pl_50_pc_2](https://user-images.githubusercontent.com/30048408/63338528-42a75b00-c343-11e9-96e2-2d40f46d4c7a.png)